quartz-core/src/main/java/org/quartz/CronExpression.java

```diff
@@ -321,7 +321,48 @@ public boolean isSatisfiedBy(Date date) {
 
         return ((timeAfter != null) && (timeAfter.equals(originalDate)));
     }
+
+    /**
+     * Indicates whether the given date satisfies the cron expression. Note that
+     * milliseconds are ignored, so two Dates falling on different milliseconds
+     * of the same second will always have the same result here.
+     * 
+     * @param date the date to evaluate
+     * @return a boolean indicating whether the given date satisfies the cron
+     *         expression
+     */
+    public boolean isSatisfiedByBefore(Date date) {
+        Calendar testDateCal = Calendar.getInstance(getTimeZone());
+        testDateCal.setTime(date);
+        testDateCal.set(Calendar.MILLISECOND, 0);
+        Date originalDate = testDateCal.getTime();
+        
+        testDateCal.add(Calendar.SECOND, 1);
+        
+        Date timeBefore = getTimeBefore(testDateCal.getTime());
+
+        return ((timeBefore != null) && (timeBefore.equals(originalDate)));
+    }
     
+    public boolean isSatisfied(Date date) {
+        Calendar testDateCal = Calendar.getInstance(getTimeZone());
+        testDateCal.setTime(date);
+        testDateCal.set(Calendar.MILLISECOND, 0);
+
+        Date timeBefore = getTimeBefore(testDateCal.getTime());
+        if (timeBefore == null) {
+            System.out.println("testDateCal " + testDateCal.getTime());
```

Comment:
Please remove System.out.println statement from the main code. If needed, use logger instead.